src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.Factory.cs

```diff
@@ -104,5 +115,26 @@ internal partial class ImmutabilityContext {
 			return new ImmutabilityContext( builder.ToImmutable() );
 		}
 
+		private static ImmutableDictionary<string, IAssemblySymbol> GetCompilationAssemblies( Compilation compilation ) {
+			var builder = ImmutableDictionary.CreateBuilder<string, IAssemblySymbol>();
+
+			builder.Add( compilation.Assembly.Name, compilation.Assembly );
+
+			foreach( MetadataReference reference in compilation.References ) {
+				switch( compilation.GetAssemblyOrModuleSymbol( reference ) ) {

```

Comment:
`Gets the IAssemblySymbol or IModuleSymbol for a metadata reference used to create this compilation.`

`Assembly or module symbol corresponding to the given reference or null if there is none.`